README.md

```diff
@@ -1 +1,102 @@
-# sdk-diatheke
\ No newline at end of file
+# SDK for Diatheke (Cobalt's Dialog Management Engine)
+
+This repository contains the SDK for Cobalt's Diathele Dialog Management Engine.
+
+This README has instructions to _build_ the SDK.  For installing and using the
+SDK, see the [SDK Docs](https://cobaltspeech.github.io/sdk-diatheke).
+
+## Network API (using GRPC)
+
+The `grpc` folder at the top level of this repository contains code for Diatheke's
+GRPC API.  The `grpc/diatheke.proto` file is the authoritative service definition of
+the API and is used for auto generating SDK code in multiple languages.
+
+### Auto-generated code and documentation
+
+The `grpc` folder contains auto-generated code in several languages.  In order
+to generate the code again, you should run `make`.  Generated code is checked
+in, and you must make sure it is up to date when you push commits to this
+repository.
+
+Code generation has the following dependencies:
+  - The protobuf compiler itself (protoc)
+  - The protobuf documentation generation plugin (protoc-gen-doc)
+  - The golang plugins (protoc-gen-go and protoc-gen-grpc-gateway)
+  - The python plugins (grpcio-tools and googleapis-common-protos)
+  - The static website generator (hugo)
+
+A few system dependencies are required:
+  - Go >= 1.12
+  - git
+  - python3
+  - virtualenv
+  - unzip
+  - wget
+
+The top level Makefile can set up all other dependencies.
+
+Documentation is authored in the `docs-src` folder and generated static website
+is stored in the `docs` folder.
+
+To generate the code and documentation, run `make`.  This is currently only
+supported under linux.
+
+If you are doing local development on the docs, you can use this command to
+serve it locally:
+
+```
+cd docs-src
+../deps/bin/hugo server -D
+```
+
+### Tagging New Versions
+
+This repository has several components, and they need more than just a "vX.Y.Z"
+tag on the git repo.  In particular, this repository has two go modules, one of
```

Comment:
Is this referring to the examples go module and the gRPC go module? If so, we will need to remember to update this section when the examples are put in their own repository.

Reply:
Honestly, I didn't look deeper on what is in other language modules. As I know, currently we don't have separate examples repos for Diatheke like for Cubic. Please, make a suggestion what else I should add to make version tagging work properly.

Reply:
My preference would be to move the example code and only have a single version on this repo. Perhaps @renaissanceGeek or @happyalu have opinions?

Reply:
I think that makes sense. It's better for the SDK to have a single version for all languages.  Ideally, when we tag a new SDK version, CI should compile the examples and flag breaking changes.